sematic/api/endpoints/runs.py

```diff
@@ -375,6 +376,17 @@ def update_run_status_endpoint(user: Optional[User]) -> flask.Response:
             )
         )
 
+    for root_id in modified_root_runs:
+        # modified_root_runs will contain only
+        # 0 or 1 root id, unless the caller asked
+        # for updates about runs in multiple
+        # resolutions at once.
+
+        # Done outside the for loop over db_status_dict.items()
+        # to minimize broadcasts for high fan-out pipelines where
+        # many runs from one root pipeline may happen at once.
+        broadcast_graph_update(root_id, user=user)
```

Comment:
The logic could be simplified if you have a single value `broadcast_to_root_id = None`, and set it to `run.root_id` instead of adding to a set.

Then
```python
if broadcast_to_root_id is not None:
    broadcast_graph_update(root_id, user=user)
```
    
Then you also don't need the lengthy comment, since the code is self-explanatory.

Reply:
The code would be *simpler* but it wouldn't be *correct*, unless we add a requirement for this endpoint that all runs come from the same root id. Unfortunately that would not be enforceable unless we always loaded the root id of every run whose state was requested and checked it. At which point *that* code is more complicated that just using a set for `modified_root_runs`.

Reply:
I'm confused. How is it possible that this end-point gets called for runs from multiple root runs?